macros_doc.md

```diff
@@ -38,10 +92,105 @@ rest of the info it should use to do the rest of its work.
 
 The only exception to this is when you want to add new content. Say you want to
 add a completely new piece of metadata to the specification. In that case, you
-will need to create the input for the macro to be able to do its job. For that,
-you will need to update the schema itself.
+will need to create the input for the macro to be able to do its job.
+
+## Why use macros at all?
```

Comment:
I think lines 97 to 141 should be moved to the end of the document and probably ultimately scrapped... but there might be some good nuggets to extract.

Reply:
Soft push back on this. I think that giving some explanation as to the "why" of some infrastructure / design choice is important.